src/Nri/Ui/Table/V8.elm

```diff
@@ -0,0 +1,368 @@
+in the table
+-}
+type Column data msg
+    = Column (Html msg) (data -> Html msg) Style (data -> List Style) (Maybe SortDirection) CellType Bool
+
+
+{-| Which direction is a table column sorted? Only set these on columns that
+actually have an explicit sort!
+-}
+type SortDirection
+    = Ascending
+    | Descending
+
+
+{-| Is this cell a data cell or header cell?
+-}
+type CellType
+    = RowHeaderCell
+    | DataCell
+
+
+cell : CellType -> List (Attribute msg) -> List (Html msg) -> Html msg
+cell cellType attrs =
+    case cellType of
+        RowHeaderCell ->
+            th (Attributes.scope "row" :: attrs)
+
+        DataCell ->
+            td attrs
+
+
+{-| A column that renders some aspect of a value as text
+-}
+string :
+    { header : String
+    , value : data -> String
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+string { header, value, width, cellStyles, sort } =
+    Column (Html.text header) (value >> Html.text) (Css.width width) cellStyles sort DataCell False
+
+
+{-| Creates a placeholder column which will reserve the space for the column,
+this can be used when multiple tables have similar data and we want to keep
+the size consistent.
+-}
+placeholderColumn :
+    { width : LengthOrAuto compatible
+    }
+    -> Column data msg
+placeholderColumn options =
+    Column (Html.text "") (always (Html.text "")) (Css.width options.width) (always []) Nothing DataCell False
+
+
+{-| A column that renders however you want it to
+-}
+custom :
+    { header : Html msg
+    , view : data -> Html msg
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+custom options =
+    Column options.header options.view (Css.width options.width) options.cellStyles options.sort DataCell True
+
+
+{-| A column whose cells are row headers
+-}
+rowHeader :
+    { header : Html msg
+    , view : data -> Html msg
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+rowHeader options =
+    Column options.header options.view (Css.width options.width) options.cellStyles options.sort RowHeaderCell False
+
+
+
+-- VIEW
+
+
+{-| Displays a table of data without a header row
+-}
+viewWithoutHeader : { additionalStyles : List Style, alternatingRowColors : Bool } -> List (Column data msg) -> List data -> Html msg
+viewWithoutHeader { additionalStyles, alternatingRowColors } columns =
+    tableWithoutHeader additionalStyles columns (viewRow columns alternatingRowColors)
+
+
+{-| Displays a table of data based on the provided column definitions
+-}
+view : { additionalStyles : List Style, alternatingRowColors : Bool } -> List (Column data msg) -> List data -> Html msg
+view { additionalStyles, alternatingRowColors } columns =
```

Comment:
💭 Since we are introducing a new version of the component, did you consider switching it to using attributes like we do in many other modules? Those are a bit more upgrade, friendly. So we could have two to start with:

```elm
css : List Style -> Attribute

-- defaults to True, which is the current behavior
-- could also be `noAlternatingRowColors : Attribute`
alternatingRowColors : Bool -> Attribute
```

Might want to check with the frontend guild before doing it, but I think it might be nice! If we already had this for table, this could have been a minor change to the module without forcing us to introduce a new version.

Reply:
OH, a bit like what you did in the SortableTable wrapper! I'm reviewing commit-by-commit leaving comments as I go :-P

Reply:
Yes, I thought about it, and I actually moved to that at some point, but it was a little strange the API to communicate between modules, but thinking about it again I think would make sense. I will change it.